types/infohash/infohash.go

```diff
@@ -46,7 +46,8 @@ func (t *T) FromHexString(s string) (err error) {
 		return
 	}
 	if n != Size {
-		panic(n)
+		err = fmt.Errorf("decoded hash hex string has bad length: %d", n)
```

Comment:
probably initially gave the right length, but the string was garbage.

Reply:
@azr what input string was able to cause the panic? I added a fuzz test to try to find anything and I'm not having any luck.

Reply:
Bummer. I'll leave it as is, if you encounter the problem again, let me know.